csproj: Target/Exec regression in VS 2019

### Steps to reproduce

Extract [CSharpExec.zip](https://github.com/Microsoft/msbuild/files/2932260/CSharpExec.zip).

At a VS 2019 RC x64 command prompt with the working directory set to the extracted folder:

```console
>msbuild CSharpExec.csproj -p:Configuration=Debug -p:VisualStudioVersion=16.0
```

Observe the build failure.

### Expected  behavior

It is expected to build successfully, as it does at a VS 2017 x64 command prompt:

```console
>msbuild -version
...
15.9.21.664

>msbuild CSharpExec.csproj -p:Configuration=Debug -p:VisualStudioVersion=15.0
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
...(no errors)...
```

### Actual behavior

The build fails at a VS 2019 RC x64 command prompt:

```console
>msbuild -version
...
16.0.443.30039

>msbuild CSharpExec.csproj -p:Configuration=Debug -p:VisualStudioVersion=16.0
Microsoft (R) Build Engine version 16.0.443+g5775d0d6bb for .NET Framework
...
  echo Generating test.cs
  Generating test.cs
  setlocal copy test.cs.in test.cs ...
  Invalid parameter to SETLOCAL command
  ...
C:\...\CSharpExec.csproj(44,5): error MSB3073: The command "setlocal copy test.cs.in test.cs ..." exited with code 1.
```

--- a/src/Build/Xml/XmlReaderExtension.cs
+++ b/src/Build/Xml/XmlReaderExtension.cs
@@ -80,21 +80,10 @@ namespace Microsoft.Build.Internal
 
             XmlReader reader;
 
-            if (loadAsReadOnly)
-            {
-                XmlReaderSettings xrs = new XmlReaderSettings
-                {
-                    DtdProcessing = DtdProcessing.Ignore,
-                    IgnoreComments = true,
-                    IgnoreWhitespace = true,
-                };
-                reader = XmlReader.Create(input, xrs, uri);
-            }
-            else
-            {
-                reader = new XmlTextReader(uri, input) { DtdProcessing = DtdProcessing.Ignore };
-            }
-
+            // Ignore loadAsReadOnly for now; using XmlReader.Create results in whitespace changes
+            // of attribute text, specifically newline removal.
+            // https://github.com/Microsoft/msbuild/issues/4210
+            reader = new XmlTextReader(uri, input) { DtdProcessing = DtdProcessing.Ignore };
 
             reader.Read();
             encoding = input.CurrentEncoding;

--- a/src/Tasks.UnitTests/Exec_Tests.cs
+++ b/src/Tasks.UnitTests/Exec_Tests.cs
@@ -5,13 +5,14 @@ using System;
 using System.IO;
 using System.Reflection;
 using System.Text;
-using Microsoft.Build.UnitTests;
+using Microsoft.Build.Execution;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Tasks;
 using Microsoft.Build.Utilities;
 using Microsoft.Build.Shared;
 using Shouldly;
 using Xunit;
+using Xunit.Abstractions;
 
 namespace Microsoft.Build.UnitTests
 {
@@ -20,9 +21,16 @@ namespace Microsoft.Build.UnitTests
     /// </summary>
     sealed public class Exec_Tests
     {
+        private readonly ITestOutputHelper _output;
+
+        public Exec_Tests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
         private Exec PrepareExec(string command)
         {
-            IBuildEngine2 mockEngine = new MockEngine(true);
+            IBuildEngine2 mockEngine = new MockEngine(_output);
             Exec exec = new Exec();
             exec.BuildEngine = mockEngine;
             exec.Command = command;
@@ -31,7 +39,7 @@ namespace Microsoft.Build.UnitTests
 
         private ExecWrapper PrepareExecWrapper(string command)
         {
-            IBuildEngine2 mockEngine = new MockEngine(true);
+            IBuildEngine2 mockEngine = new MockEngine(_output);
             ExecWrapper exec = new ExecWrapper();
             exec.BuildEngine = mockEngine;
             exec.Command = command;

--- a/src/Tasks.UnitTests/Exec_Tests.cs
+++ b/src/Tasks.UnitTests/Exec_Tests.cs
@@ -13,6 +13,8 @@ using Microsoft.Build.Shared;
 using Shouldly;
 using Xunit;
 using Xunit.Abstractions;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
 
 namespace Microsoft.Build.UnitTests
 {
@@ -848,6 +850,57 @@ namespace Microsoft.Build.UnitTests
 
             Assert.False(Exec.CanEncodeString(defaultEncoding.CodePage, nonAnsiCharacters));
             Assert.True(Exec.CanEncodeString(defaultEncoding.CodePage, pathWithAnsiCharacters));
+        }
+
+        [Fact]
+        public void EndToEndMultilineExec()
+        {
+            using (var env = TestEnvironment.Create(_output))
+            {
+                var testProject = env.CreateTestProjectWithFiles(@"<Project>
+ <Target Name=""MultilineExec"">
+  <Exec Command=""echo line 1
+echo line 2
+echo line 3"" />
+   </Target>
+</Project>");
+
+                using (var buildManager = new BuildManager())
+                {
+                    MockLogger logger = new MockLogger(_output, profileEvaluation: false, printEventsToStdout: false);
+                    var parameters = new BuildParameters()
+                    {
+                        Loggers = new[] { logger },
+                    };
+
+                    var collection = new ProjectCollection(
+                        new Dictionary<string, string>(),
+                        new[] { logger },
+                        remoteLoggers: null,
+                        ToolsetDefinitionLocations.Default,
+                        maxNodeCount: 1,
+                        onlyLogCriticalEvents: false,
+                        loadProjectsReadOnly: true);
+
+                    var project = collection.LoadProject(testProject.ProjectFile).CreateProjectInstance();
+
+                    var request = new BuildRequestData(
+                        project,
+                        targetsToBuild: new[] { "MultilineExec" },
+                        hostServices: null);
+
+                    var result = buildManager.Build(parameters, request);
+
+                    logger.AssertLogContains("line 2");
+                    logger.AssertLogContains("line 3");
+
+                    // To be correct, these need to be on separate lines, not
+                    // all together on one.
+                    logger.AssertLogDoesntContain("1 echo line");
+
+                    result.OverallResult.ShouldBe(BuildResultCode.Success);
+                }
+            }
         }
     }
 

